File: src/main/java/com/ven2see/springth2see/model/Post.java
package com.ven2see.springth2see.model;

import org.springframework.web.multipart.MultipartFile;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import lombok.Data;

@Data
@Entity
@Table(name="post")
public class Post {
     @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private int id;
     private String title;
     private String brief;
     private String content;
     private String image;
     @Transient // Esta anotación indica que este campo no se debe persistir en la base de datos
     private MultipartFile file; 
     private int status;
}

File: springth2see/frontview1.0/src/app/model/Post.ts
import { User } from "./User";

export class Post {
  id:number=0;
  title: string='';
  brief: string='';
  content: string='';
  image: string='';
  status: number=0;
  user: User = new User();
  created_at: string='';
  category: number=0;
  favoriteBy: User[]=[];
  liked: boolean=false;
}

